crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,156 @@
+use tempfile::{tempdir, TempDir};
+
+#[derive(Default)]
+pub struct OctezClientBuilder {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    // if None, use localhost:8732 (the default endpoint for octez-node)
+    endpoint: Option<OctezNodeEndpoint>,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientBuilder {
+    pub fn new() -> Self {
+        OctezClientBuilder::default()
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = Some(endpoint);
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
+        OctezClient {
+            binary_path: self.binary_path.take(),
+            base_dir,
+            endpoint: self.endpoint.unwrap_or_default(),
+            disable_unsafe_disclaimer: self.disable_unsafe_disclaimer,
+        }
+    }
+}
+
+pub enum OctezNodeEndpoint {
+    RpcEndpoint(Endpoint),
+    Node(OctezNode),
+}
+
+impl Default for OctezNodeEndpoint {
+    fn default() -> Self {
+        OctezNodeEndpoint::RpcEndpoint(Endpoint::localhost(8732))
+    }
+}
+
+impl From<Endpoint> for OctezNodeEndpoint {
+    fn from(endpoint: Endpoint) -> Self {
+        OctezNodeEndpoint::RpcEndpoint(endpoint)
+    }
+}
+impl From<OctezNode> for OctezNodeEndpoint {
+    fn from(node: OctezNode) -> Self {
+        OctezNodeEndpoint::Node(node)
+    }
+}
+
+pub enum OctezClientDirectory {
+    TempDir(TempDir),
+    Path(PathBuf),
+}
+
+impl Default for OctezClientDirectory {
+    fn default() -> Self {
+        OctezClientDirectory::TempDir(tempdir().expect("Failed to create tempdir"))
+    }
+}
+
+// TODO: remove later
+#[allow(dead_code)]
+pub struct OctezClient {
+    // if None, use the binary in $PATH
+    binary_path: Option<PathBuf>,
+    base_dir: OctezClientDirectory,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    #[test]
+    fn builds_default_octez_client() {
+        let octez_client = OctezClientBuilder::new().build();
+        assert!(octez_client.binary_path.is_none());
+        assert!(matches!(
+            octez_client.base_dir,
+            OctezClientDirectory::TempDir(_)
+        ));
+        assert!(!octez_client.disable_unsafe_disclaimer);
+        match octez_client.endpoint {
+            OctezNodeEndpoint::RpcEndpoint(endpoint) => {
```

Comment:
I suggest to implement derive `PartialEq` on OctezNodeEndpoint, then you could do
`assert_eq!(octez_client.endpoint, OctezNodeEndpoint::RpcEndpoint(Endpoint::localhost(8732)))`